### PR TITLE
add extra helper context wrapper for getting project and cluster

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- Temporary solution to avoid conditionals checking if currentProject / currentCluster / user are defined or not

### CODE REVIEW

File: dashboard/src/main/auth/context.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  currentProject: ProjectType;
+  currentCluster: ClusterType;
+  user?: PorterUser;
+};
+
+export const AuthStateContext = createContext<AuthContextType | null>(null);
+
+export const useAuthState = (): AuthContextType => {
+  const context = useContext(AuthStateContext);
+  if (context === null) {
```
Comment (ferozemohideen) [inline dashboard/src/main/auth/context.tsx:23]:
```suggestion
  if (context == null) {
```

File: dashboard/src/main/auth/context.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  user?: PorterUser;
+};
+
+export const AuthStateProvider: React.FC<AuthStateProviderProps> = ({
+  children,
+  currentProject,
+  currentCluster,
+  user,
+}) => {
+  if (!currentProject || !currentCluster || !user) {
```
Comment (ferozemohideen) [inline dashboard/src/main/auth/context.tsx:42]: maybe add checks for > 0?

File: dashboard/src/main/auth/context.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import { type ClusterType, type ProjectType } from "shared/types";
+
+type PorterUser = {
+  userId: number;
+  email: string;
+  isPorterUser: boolean;
+};
+
+type AuthContextType = {
+  currentProject: ProjectType;
```
Comment (ferozemohideen) [inline dashboard/src/main/auth/context.tsx:14]: should we call these project and cluster to differentiate from the old context? maybe even clientProject and clientCluster? idk

**Review by ferozemohideen — APPROVED** [review]

File: dashboard/src/main/auth/context.tsx (hunk 3 above)
Comment (ianedwards) [reply dashboard/src/main/auth/context.tsx:14]: yeah fair point

File: dashboard/src/main/auth/context.tsx (hunk 2 above)
Comment (ianedwards) [reply dashboard/src/main/auth/context.tsx:42]: I'm gonna get rid of the default values so this shouldn't apply anymore

File: dashboard/src/main/auth/context.tsx (hunk 1 above)
Comment (ianedwards) [reply dashboard/src/main/auth/context.tsx:23]: I think we actually have a lint rule that prefers the former: https://eslint.org/docs/latest/rules/eqeqeq

**Pull request merged**